### PR TITLE
New Naval Professions, mended PR

### PR DESCRIPTION
**Pull request by DoctorBoomstick**
#### Summary
Content "Adds four new naval professions to the aircraft carrier scenario."
Purpose of change
I felt like the aircraft carrier scenario could do with a little uniqueness in its available professions, rather than the stock standard military starts that are currently present. In addition, this PR tries to add more realism to the sorts of jobs that the player could have realistically been engaged in whilst abord an aircraft carrier that was on active deployment. To my knowledge, and according to a few searches that I did, regular army infantry are normally not present abord Naval ships when said vessels are on assignment.
Describe the solution
This PR adds four new Navy based professions, and removes the ones which I thought were the most unlikely to be actually abord a carrier. I’ll probably add a few more down the line, such as an aircraft technician and communications officer, but these are the ones I judged should be added first.
I’ve implemented the Navy Sailor, which replaces the former military recruit and combat mechanic professions. This job starts with basic military style clothing, a few tools, a wide portion of the mechanical and metal working proficiencies, moderate skill in mechanics and athletics, and no weapons apart for a really, really big wrench.
The rifleman profession has been changed to the Marine Infantryman, under the notion that most ground forces that would be abord a carrier would more than likely be marines or Navy special warfare personnel. This profession has skills in rifles, marksmanship, Melee, pistols and athletics, and starts with standard military clothing and armour, as well as an Mk 18 and M18, with two respective magazines for each.
I’ve added the Navy SEAL start as a replacement for the special operator, in order to reflect the Navy’s own special warfare contingent that are a different outfit from the army’s special forces units. This profession starts with higher level military clothing and Armor, advanced electronics such as shooter’s earmuffs and light amplification goggles, high to moderate skills in rifles, marksmanship, athletics, melee, pistols, and stabbing, and an Mk 18 and P226 MK25 as firearms, plus their respective magazines.
Lastly, I added the Navy Diver profession as a bit of flavour and variety to the available professions. Starting with cold weather diving gear, the profession has high skills in athletics and moderate skills in survival, marksmanship and healthcare, with a diving knife, heavy duty flashlight, and a few gadgets like a barometer and rebreather.
I’ve also added the eight point cap, the standard cap of the marines and navy, to the existing line-up of hats, and equipped the sailor and marine starts with it.
Describe alternatives you've considered
Adding naval zombies first and dealing with the professions later.
Testing
Spawned in with all the professions and checked that all their gear functioned as intended.
Additional context
https://www.quora.com/What-does-a-Navy-Seals-uniform-look-like
https://www.operationmilitarykids.org/navy-uniform-regulations/#:~:text=The%20Navy%20working%20uniform%20is%20their%20version%20of,doing%20all%20Navy%20tasks%20during%20assigned%20duty%20hours.
https://en.wikipedia.org/wiki/Uniforms_of_the_United_States_Marine_Corps

### CODE REVIEW

**Comment by DoctorBoomstick** [comment]
The last PR that I made of this accidentally had another unfinished PR mashed in along with it. this fixes that and removes the accidentally added other files.

File: data/json/professions.json [hunk 1]
```diff
@@ -911,6 +923,145 @@
...
+      "male": [ "boxer_shorts" ],
+      "female": [ "sports_bra", "boxer_shorts" ]
+    },
+    "flags": [ "SCEN_ONLY" ]
+  },
+  {
+    "type": "profession",
+    "id": "diver_navy",
+    "name": "Navy Diver",
+    "description": "Before the cataclysm, you used to be part of a Naval Special Warfare diving group, a skilled soldier adept at underwater infiltration and marine combat.  You were just finishing suiting up for a practice dive before the screams interrupted your preparations. Now, you’ve got nothing to ward off your former crew-mates except for your wetsuit, equipment, and training.",
```
Comment (GuardianDll) [inline data/json/professions.json:1029]:
```suggestion
    "description": "Before the cataclysm, you used to be part of a Naval Special Warfare diving group, a skilled soldier adept at underwater infiltration and marine combat.  You were just finishing suiting up for a practice dive before the screams interrupted your preparations.  Now, you’ve got nothing to ward off your former crew-mates except for your wetsuit, equipment, and training.",
```

**Review by TheShadowFerret — COMMENTED** [review]
Note: I don't remember if Cataclysm is supposed to be capitalized or not

File: data/json/items/armor/hats.json [hunk 2]
```diff
@@ -174,6 +174,25 @@
     "environmental_protection": 1,
     "armor": [ { "encumbrance_modifiers": [ "NONE" ], "coverage": 60, "covers": [ "head" ] } ]
   },
+  {
+    "id": "hat_navy",
+    "type": "ARMOR",
+    "name": { "str": "Eight point cap" },
+    "description": "Also known as a utility cover or utility cap, this is an eight pointed hat with a baseball style vizor at the front.  Commonly seen worn by Marines and sailors, this hat is the standard headwear of both the United States Marine Corps and Navy.",
```
[the remark this answers is not in the document]
Comment (NetSysFire) [reply data/json/items/armor/hats.json:181]: >Note: I don't remember if Cataclysm is supposed to be capitalized or not

Cataclysm should be capitalized if the Cataclysm is meant. If it is a cataclysm, not the event the game is about, you do not capitalize it. In this case, it should be capitalized.

**Comment by NetSysFire** [comment]
You still need to capitalize Cataclysm in your descriptions

File: data/json/professions.json [hunk 3]
```diff
@@ -911,6 +923,145 @@
...
+      "male": [ "boxer_shorts" ],
+      "female": [ "sports_bra", "boxer_shorts" ]
+    },
+    "flags": [ "SCEN_ONLY" ]
+  },
+  {
+    "type": "profession",
+    "id": "diver_navy",
+    "name": "Navy Diver",
+    "description": "Before the Cataclysm, you used to be part of a Naval Special Warfare diving group, a skilled soldier adept at underwater infiltration and marine combat.  You were just finishing suiting up for a practice dive before the screams interrupted your preparations. Now, you’ve got nothing to ward off your former crew-mates except for your wetsuit, equipment, and training.",
```
Comment (GuardianDll) [inline data/json/professions.json:1029]:
```suggestion
    "description": "Before the Cataclysm, you used to be part of a Naval Special Warfare diving group, a skilled soldier adept at underwater infiltration and marine combat.  You were just finishing suiting up for a practice dive before the screams interrupted your preparations.  Now, you’ve got nothing to ward off your former crew-mates except for your wetsuit, equipment, and training.",
```

**Pull request merged**